sema/subyt/sources.py

```
@@ -284,15 +317,30 @@ class CSVFileSource(Source):
+        def __init__(self, csv_file_path: Path, config: dict = {}) -> None:
             super().__init__()
             assert_readable(csv_file_path)
             self._csv: Path = csv_file_path.absolute()
             self._init_source(self._csv)
+            self._csvconfig: dict = dict()
+            for key in ["delimiter", "quotechar", "header"]:
+                if key in config:
+                    self._csvconfig[key] = config[key]
 
         def __enter__(self) -> object:
             self._csvfile = open(self._csv, mode="r", encoding="utf-8-sig")
```

Ruff (0.8.2)

Unnecessary open mode parameters

Remove open mode parameters

(UP015)